addons/udes_stock/tests/test_product_template.py

```diff
@@ -0,0 +1,26 @@
+"""Unit tests for the ProductTemplate model."""
+
+from odoo.exceptions import ValidationError
+
+from . import common
+
+
+class ProductTemplateDeletionTestCase(common.BaseUDES):
+    """Tests for prevention of deletion of product templates."""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.apple_template = cls.apple.product_tmpl_id
+        cls.red_apple = cls.apple
+
+    def test_raises_error_on_deletion(self):
+        """Unlinking a product template should raise an error."""
+        expected_message = "Products may not be deleted. Please archive them instead."
```

Comment:
Nit: I tried to make a suggestion the only one though maybe was to make the expected message a constant and to import in unit tests. In case that the message will change no need to change it in unit tests too.

Reply:
I think it's better that the message is defined separately in the tests.  That way we can be sure that any change to the message is definitely intentional.

Reply:
Agree, thats happen when i hardly try to add a thread :)